core/domain/story_services.py

```diff
@@ -617,3 +621,258 @@ def record_completed_node_in_story_context(user_id, story_id, node_id):
+        Exception. The user does not have enough rights to unpublish the story.
+    """
+    story_rights = get_story_rights(story_id, strict=False)
+    if story_rights is None:
+        raise Exception('The given story does not exist')
+    user = user_services.UserActionsInfo(committer_id)
+    if role_services.ACTION_CHANGE_STORY_STATUS not in user.actions:
+        raise Exception(
+            'The user does not have enough rights to unpublish the story.')
+
+    if not story_rights.story_is_published:
+        raise Exception('The story is already unpublished.')
+    story_rights.story_is_published = False
+    commit_cmds = [story_domain.StoryRightsChange({
+        'cmd': story_domain.CMD_UNPUBLISH_STORY
+    })]
+    save_story_rights(
+        story_rights, committer_id, 'Unpublished the story', commit_cmds)
+
+
+def save_story_rights(story_rights, committer_id, commit_message, commit_cmds):
+    """Saves a StoryRights domain object to the datastore.
+
+    Args:
+        story_rights: StoryRights. The rights object for the given
+            story.
+        committer_id: str. ID of the committer.
+        commit_message: str. Descriptive message for the commit.
+        commit_cmds: list(StoryRightsChange). A list of commands describing
+            what kind of commit was done.
+    """
+
+    model = story_models.StoryRightsModel.get(story_rights.id, strict=False)
+
+    model.manager_ids = story_rights.manager_ids
+    model.story_is_published = story_rights.story_is_published
+    commit_cmd_dicts = [commit_cmd.to_dict() for commit_cmd in commit_cmds]
+    model.commit(committer_id, commit_message, commit_cmd_dicts)
+
+
+def create_new_story_rights(story_id, committer_id):
+    """Creates a new story rights object and saves it to the datastore.
+
+    Args:
+        story_id: str. ID of the story.
+        committer_id: str. ID of the committer.
+    """
+    story_rights = story_domain.StoryRights(story_id, [], False)
+    commit_cmds = [{'cmd': story_domain.CMD_CREATE_NEW}]
+
+    story_models.StoryRightsModel(
+        id=story_rights.id,
+        manager_ids=story_rights.manager_ids,
+        story_is_published=story_rights.story_is_published
+    ).commit(committer_id, 'Created new story rights', commit_cmds)
+
+
+def get_story_rights(story_id, strict=True):
+    """Retrieves the rights object for the given story.
+
+    Args:
+        story_id: str. ID of the story.
+        strict: bool. Whether to fail noisily if no story with a given id
+            exists in the datastore.
+
+    Returns:
+        StoryRights. The rights object associated with the given story.
+
+    Raises:
+        EntityNotFoundError. The story with ID story_id was not
+            found in the datastore.
+    """
+
+    model = story_models.StoryRightsModel.get(story_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_story_rights_from_model(model)
+
+
+def get_all_story_rights():
```

Comment:
Is this function used / necessary?

Reply:
When I think of it, no we don't actually use this.